modal/cls.py

```diff
@@ -395,6 +395,7 @@ def _initialize_from_empty(self):
 
     def _initialize_from_other(self, other: "_Cls"):
         super()._initialize_from_other(other)
+        self._app = other._app
```

Comment:
Any downsides to this? Not sure why we didn't have it previously ... makes me a little nervous.

Reply:
I *think* it should be fine. I really dislike how many member vars we have on classes and functions and I think we should try to group them into distinct sets based on which are expected to be set at any given time. Would make it so much easier to reason about these things

Reply:
Yes plz